asus/battery.nix

```
@@ -0,0 +1,21 @@
+{ config, pkgs, lib, ... }:
+let
+  p = pkgs.writeScriptBin "charge-upto" ''
+    echo ''${0:-100} > /sys/class/power_supply/BAT0/charge_control_end_threshold
+  '';
+  cfg = config.hardware.asus;
+in
+
+{
+  options.hardware.asus.chargeUpto = lib.mkOption {
```

In case we get more options in future:

```suggestion
  options.hardware.asus.battery.chargeUpto = lib.mkOption {
```